File: Decorator/C#/CoffeeShopDecorator/WhippedCreamCondiment.cs
namespace CoffeeShopDecorator
{
    //ConcreteDecorator
    public class WhippedCreamCondiment : CondimentDecorator
    {
        public WhippedCreamCondiment(Beverage beverage) : base(beverage)
        {
            Description = "whipped cream";
        }

        public override string GetDescription()
        {
            return $"{base.GetDescription()}, {Description}";
        }

        public override decimal GetPrice()
        {
            return base.GetPrice() + 0.4m;
        }

    }
}

File: Decorator/Java/CoffeeShowDecorator/src/CoffeeShopDecorator/WhippedCreamCondiment.java
package CoffeeShopDecorator;

//ConcreteDecorator
public class WhippedCreamCondiment extends CondimentDecorator
{
    public WhippedCreamCondiment(Beverage beverage) 
    {
    	super(beverage);
        description = "whipped cream";
    }

    @Override
    public double getPrice()
    {
        return wrappedBeverage.getPrice() + 0.4d;
    }

    @Override
    public String getDescription()
    {
        return String.format("%s, %s", wrappedBeverage.getDescription(), description);
    }
}